### ISSUE TITLE
Word Count is including HTML tags

### ISSUE THREAD

**Issue by koke** [wordpress-mobile/WordPress-iOS#13618]
It looks like the code to count words is not stripping HTML tags, which means it will overcount words on Gutenberg posts.

To reproduce:

1. Watch the logs for the `editor_post_published` event, it should have a `word_count` property.
2. Set your editor to classic and publish a post with just the word `Hello`: word count shows 1
3. Set your editor to block and  publish a post with just the word `Hello`: word count shows 6

**Comment by etoledom** [comment]
This also happens on Aztec posts, testing with "hello **world**" (world in Bold), it will count 4 words.

The logic will count words from the post content, whatever it is:
https://github.com/wordpress-mobile/WordPress-iOS/blob/cdff8dc6d5b28ceede45c6a7a261d666f3d52803/WordPress/Classes/ViewRelated/Post/PostEditor%2BPublish.swift#L167

**Comment by koke** [comment]
For reference, Android seems to convert the HTML to plain text first:

https://github.com/wordpress-mobile/WordPress-Android/blob/0f7f9cf3f18eb81aef4c883bdea0b53b88a187fe/WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java#L163-L166

**Comment by etoledom** [comment]
So, I had the idea of using `NSAttributedString` conversion to strip out HTML code, to be sure we are doing it right, so I did some tests against Regex.

As expected, `NSAttributedStrings` are much slower, but also the words count are different, with Regex counting more words.

Testing scenario:
Post of 25464 words (counted by Gutenberg Web)

**NSAttributedString:**
- 3~4 seconds 
- 27200 words counted

**Regex `<[^>]+>`** (Mentioned in a couple of sites)
- 0.01~ seconds
- 27476 words counted

**Regex `<img[^>]*>`** (From Android code)
- 0.01~ seconds
- 38184 words counted


The web uses a fairly more complex algorithm, still based on RegEx. We could reuse this library on mobile, but we would still need to do something on Aztec (And hopefully make both counts match)

https://github.com/WordPress/gutenberg/blob/master/packages/wordcount/src/index.js#L123

As a fast hotfix, I would go with matching Android, but that solution might also need a revision.

**Comment by koke** [comment]
Gutenberg is also more advanced in that it supports non-word based counts for some languages:
https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/word-count/index.js#L10

**Comment by maxme** [comment]
> Regex <img[^>]*> (From Android code)

Note the code on Android:
1. strips `img` tags. 
2. converts this to text (`Html.fromHtml` drops all html tags).

I think there is a specific case for `img` because `Html.fromHtml` d would add an "object" unicode character and that would count for one word.

**Comment by etoledom** [comment]
I copied [this](https://medium.com/linux-plus-plus/linux-march-8-2020-d851f5ff8882) post and pasted it 5 or 6 times to test speed. And [this is](https://mysitefortestingetoledom.wordpress.com/2020/03/10/1689/) the final result.

I had a problem on Gutenberg web where I couldn't paste the whole content (it got cropped at the first image) so I pasted it to Notes app first. I noticed now that Notes app created some extra paragraphs after the images, and those were converted to paragraph blocks like this:

```html
<!-- wp:paragraph -->
<p>&lt;img class="cx t u hi ak" src="https://miro.medium.com/max/9376/1*06JfyotscxggMR0RFK1POA.png" width="4688" height="1563" role="presentation"/&gt;</p>
<!-- /wp:paragraph -->
```

This might be a cause of the different word count 🤔
